src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/PropertySetAnalysis/PropertySetCallbacks.cs

```diff
@@ -0,0 +1,177 @@
+                    }
+                    else
+                    {
+                        // We don't know all values but didn't find any bad value, so we can say who knows.
+                        return PropertySetAbstractValueKind.Unknown;
+                    }
+
+                default:
+                    return PropertySetAbstractValueKind.Unknown;
+            }
+        }
+
+        /// <summary>
+        /// A <see cref="HazardousUsageEvaluator.EvaluationCallback"/> for all properties flagged being hazardous.
+        /// </summary>
+        /// <param name="propertySetAbstractValue">PropertySetAbstract value.</param>
+        /// <returns>If all properties are flagged, then flagged; if at least one property is unflagged, then unflagged; otherwise maybe flagged.</returns>
+        public static HazardousUsageEvaluationResult HazardousIfAllFlagged(PropertySetAbstractValue propertySetAbstractValue)
+        {
+            if (propertySetAbstractValue.KnownValuesCount == 0)
+            {
+                // No known values implies only PropertySetAbstractValueKind.Unknown.
+                return HazardousUsageEvaluationResult.MaybeFlagged;
+            }
+
+            bool allFlagged = true;
+            bool atLeastOneUnflagged = false;
+            for (int i = 0; i < propertySetAbstractValue.KnownValuesCount; i++)
+            {
+                if (propertySetAbstractValue[i] != PropertySetAbstractValueKind.Flagged)
+                {
+                    allFlagged = false;
+                }
+
+                if (propertySetAbstractValue[i] == PropertySetAbstractValueKind.Unflagged)
+                {
+                    atLeastOneUnflagged = true;
+                    break;
+                }
+            }
+
+            if (allFlagged)
+            {
+                return HazardousUsageEvaluationResult.Flagged;
+            }
+            else if (atLeastOneUnflagged)
+            {
+                return HazardousUsageEvaluationResult.Unflagged;
+            }
+            else
+            {
+                return HazardousUsageEvaluationResult.MaybeFlagged;
+            }
+        }
+
+        /// <summary>
+        /// A <see cref="HazardousUsageEvaluator.InvocationEvaluationCallback"/> for all properties flagged being hazardous.
+        /// </summary>
+        /// <param name="methodSymbol">Ignored.  If your scenario cares about the method, don't use this.</param>
+        /// <param name="propertySetAbstractValue">PropertySetAbstract value.</param>
+        /// <returns>If all properties are flagged, then flagged; if all properties are unflagged, then unflagged; otherwise maybe flagged.</returns>
+        [SuppressMessage("Usage", "CA1801", Justification = "Intentionally ignored; have to match delegate signature")]
+        [SuppressMessage("Usage", "IDE0060", Justification = "Intentionally ignored; have to match delegate signature")]
```

Comment:
You can suppress `IDE0060` here be naming the unused parameters as `_`, `_1`, `_2`, etc. I am going to make the same change for `CA1801`

Reply:
Sorry, not following what you mean?